data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/main/ConsumerVerticleFactoryImpl.java

```diff
@@ -148,72 +154,100 @@ public AbstractVerticle get(final DataPlaneContract.Resource resource, final Dat
-      })
-        .mapEmpty();
+       KafkaClientsAuth.attachCredentials(consumerConfigs, credentials);
+       KafkaClientsAuth.attachCredentials(producerConfigs, credentials);
+
+       final KafkaConsumer<Object, CloudEvent> consumer = createConsumer(vertx, consumerConfigs);
+       AutoCloseable metricsCloser = Metrics.register(consumer.unwrap());
+
+       final var egressConfig =
+         egress.hasEgressConfig() ?
+           egress.getEgressConfig() :
+           resource.getEgressConfig();
+
+       final var egressSubscriberSender = createConsumerRecordSender(
+         vertx,
+         egress.getDestination(),
+         egressConfig
+       );
+
+       final var egressDeadLetterSender = hasDeadLetterSink(egressConfig)
+         ? createConsumerRecordSender(vertx, egressConfig.getDeadLetter(), egressConfig)
+         : NO_DEAD_LETTER_SINK_SENDER;
+
+       final var filter = getFilter(egress);
+
+       final var responseHandler = getResponseHandler(egress,
+         () -> getResponseToKafkaTopicHandler(vertx, producerConfigs, resource),
+         () -> new ResponseToHttpEndpointHandler(createConsumerRecordSender(vertx, egress.getReplyUrl(), egressConfig)));
+       final var commitIntervalMs = Integer.parseInt(String.valueOf(consumerConfigs.get(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG)));
+
+       final var recordDispatcher = new RecordDispatcherMutatorChain(
+         new RecordDispatcherImpl(
+           new ResourceContext(resource, egress),
+           filter,
+           egressSubscriberSender,
+           egressDeadLetterSender,
+           responseHandler,
+           new OffsetManager(vertx, consumer, eventsSentCounter::increment, commitIntervalMs),
+           ConsumerTracer.create(
+             ((VertxInternal) vertx).tracer(),
+             new KafkaClientOptions()
+               .setConfig(consumerConfigs)
+               // Make sure the policy is propagate for the manually instantiated consumer tracer
+               .setTracingPolicy(TracingPolicy.PROPAGATE)
+           ),
+           Metrics.getRegistry()
+         ),
+         new CloudEventOverridesMutator(resource.getCloudEventOverrides())
+       );
+
+       // Set all the built objects in the consumer verticle
+       consumerVerticle.setRecordDispatcher(recordDispatcher);
+       consumerVerticle.setConsumer(consumer);
+       consumerVerticle.setCloser(AsyncCloseable.wrapAutoCloseable(metricsCloser));
+     })
+       .mapEmpty();
 
     return getConsumerVerticle(deliveryOrder, initializer, new HashSet<>(resource.getTopicsList()));
   }
 
+  private Filter getFilter(DataPlaneContract.Egress egress) {
+    // Dialected filters should override the attributes filter
+    if (egress.getDialectedFilterCount() > 0) {
+      logger.debug("Egress contains dialected-filters. Dialected-filters will be used for event filtering. Egress {}", egress.getReference());
+      return getFilter(egress.getDialectedFilterList());
+    } else if (egress.hasFilter()) {
```

Comment:
`else if` after `return`
```suggestion
      return getFilter(egress.getDialectedFilterList());
    }
    if (egress.hasFilter()) {
```

Reply:
I don't see why is this better than the other? care to elaborate? is it a personal preference?

Reply:
The else is useless if the prev branch has a return and this is the pattern used everywhere.

Reply:
> The else is useless if the prev branch has a return and this is the pattern used everywhere in this codebase.

If it's the pattern used everywhere then it makes sense to be consistent. Fwiw I personally prefer the if/elseif/elseif cause it gives me a better higher level mental model about the codepath without needing to dig into each branch, e.g. if the IDE is folding code blocks.

Reply:
Ok, fine for me, I see we use both :man_shrugging: